csrc/scheduler/utils.cpp

```diff
@@ -1768,8 +1768,13 @@ BroadcastMultipleInformation getBroadcastMultiples(
 
   // We always cacheBefore output at the beginning of the scheduling. And after
   // cacheBefore, the reference tensor will have all reduction IDs removed.
-  auto ref_root_domain = TensorDomain::noDevices(
-      TensorDomain::noReductions(reference_tv->getLogicalDomain()));
+  auto ref_root_domain_view = reference_tv->getLogicalDomain() |
+      TensorDomain::kNoReductions | TensorDomain::kNoDevices;
+  std::vector<IterDomain*> ref_root_domain;
```

Comment:
Why not use range constructor with std::vector?
```suggestion
  std::vector<IterDomain*> ref_root_domain(ref_root_domain_view.begin(), ref_root_domain_view.end());
```